bin/commitments_deprecater.rb

```diff
@@ -0,0 +1,190 @@
+# frozen_string_literal: true
+
+require "cluster"
+require "services"
+require "deprecation_record"
+
+Services.mongo!
+
+# Takes (shared print) deprecation records, and tries to match them to commitments
+# so that those commitments can be deprecated.
+# Usage: CommitmentsDeprecater.new.run(<list_of_files>)
+
+class CommitmentsDeprecater
+  attr_reader :err, :log_path, :report
+
+  def initialize(verbose: false)
+    @verbose = verbose
+    @header_spec = ["organization", "ocn", "local_id", "deprecation_status"]
+  end
+
+  # Process a list of files.
+  def run(args)
+    args.each do |arg|
+      if File.file?(arg)
+        from_file(arg)
+      else
+        log "arg #{arg} not a file, skipping"
+      end
+    end
+  end
+
+  # Write log message to file, and STDERR if @verbose
+  def log(msg)
+    if @report.nil?
+      # To reduce the risk of accidental overwrites by a different instance,
+      # make the filename rather unique-ish.
+      iso_stamp = Time.now.strftime("%Y%m%d-%H%M%S")
+      rand_str  = SecureRandom.hex(8)
```

Comment:
`ENV['HOSTNAME']` might work. We should also discuss the use cases for these logs & if just logging to stdout & reviewing later as needed with loki would be sufficient.

Reply:
Either way, we should probably consider logging via `Services.logger` rather than defining a custom logger here.

Reply:
These are files that we are likely to want to upload to a member box account though.

Reply:
Got it -- in that case I think the approach here makes sense; that might be worth explicitly noting.

Reply:
I've tried to reword it so it's more clear that it's a report and we are appending messages to a report, rather than logging to a log.